### PR TITLE
#3765 [fix] <fig> content is out of order in PDF

### LINKED ISSUE #3765: <fig> content is processed out of order in PDF2
## Expected Behavior
in PDF,
```
<fig>
  <title/>
  <desc/>
  <image/>
</fig>
```
is rendered as
```
<fig>
  <image/>
  <title/>
  <desc/>
</fig>
```
(in proper order)

## Actual Behavior
in PDF,
```
<fig>
  <title/>
  <desc/>
  <image/>
</fig>
```
is rendered as
```
<fig>
  <desc/>
  <image/>
  <title/>
</fig>
```
(out of order)

## Possible Solution
this part
https://github.com/dita-ot/dita-ot/blob/9d0ad7495db0240ef7acee99b3593368acd90a89/src/main/plugins/org.dita.pdf2/xsl/fo/topic.xsl#L1025
needs to carter for desc
```
            <xsl:apply-templates select="*[not(contains(@class,' topic/title ') or contains(@class,' topic/desc '))]"/>
            <xsl:apply-templates select="*[contains(@class,' topic/title ') or contains(@class,' topic/desc ')]"/>
```

### PR DESCRIPTION
**Pull request by thendarion**
Signed-off-by: thendarion <16006640+thendarion@users.noreply.github.com>

## Description
fixes #3765

## How Has This Been Tested?
unit test

### CODE REVIEW

**Comment by thendarion** [comment]
hmm... apparently the XSpec in oXygen and the XSpec in Gradle disagree about which things are missing and which aren't, but that shouldn't be too hard to fix

**Comment by thendarion** [comment]
this XSpec implementation took a while to satisfy. I hope GitHub lets us squash these commits into one :D

File: build.gradle [hunk 1]
```diff
@@ -391,6 +391,9 @@ test {
         'src/test/xsl/common/dita-utilities.xspec',
         'src/test/xsl/common/uri-utils.xspec',
         'src/test/xsl/plugins/org.dita.html5/xsl/functions.xspec',
+        'src/test/xsl/plugins/org.dita.html5/xsl/simpletable.xspec',
+        'src/test/xsl/plugins/org.dita.pdf2/xsl/fo/simpletable.xspec',
```
Comment (thendarion) [inline build.gradle:395]: these two were already there but they were not included in this file -- which I assume determines which tests to run

File: build.gradle [hunk 2]
```diff
@@ -391,6 +391,9 @@ test {
...
+        'src/test/xsl/plugins/org.dita.pdf2/xsl/fo/fig-order.xspec',
```
Comment (thendarion) [inline build.gradle:396]: this is my new file

File: src/main/plugins/org.dita.pdf2/xsl/fo/topic.xsl [hunk 3]
```diff
@@ -1022,8 +1022,8 @@ See the accompanying LICENSE file for applicable license.
                 <xsl:call-template name="get-id"/>
               </xsl:attribute>
             </xsl:if>
-            <xsl:apply-templates select="*[not(contains(@class,' topic/title '))]"/>
-            <xsl:apply-templates select="*[contains(@class,' topic/title ')]"/>
+            <xsl:apply-templates select="*[not(contains(@class,' topic/title ') or contains(@class,' topic/desc '))]"/>
+            <xsl:apply-templates select="*[contains(@class,' topic/title ') or contains(@class,' topic/desc ')]"/>
```
Comment (thendarion) [inline src/main/plugins/org.dita.pdf2/xsl/fo/topic.xsl:1026]: the structure of <fig> is title then desc then everything else, so the template needs to deal with both desc and title, not only title separately

ref: https://docs.oasis-open.org/dita/v1.0/langspec/fig.html

**Review by jelovirt — APPROVED** [review]

**Pull request merged**